### PR TITLE
Allow-Slots-Shadow-PseudoVariables 

### PR DESCRIPTION
**Pull request by MarcusDenker**
This is maybe a bit controversial:

PseudoVariables are just Variables, there is nothing that should, on the level of the class builder, raise a hard error if you want to shadow "self" with a slot named "self".

The tools will warn you that the variable is shadowing an existing variable of that name already, we could add a special rule (or improve the description) for this case in addition.

A side effect is that this means that thisProcess as a instance variable would now still be possible and thus introducing thisProcess is less of a backward compatibility problem.

### CODE REVIEW

File: src/Kernel/Slot.class.st [hunk 1]
```diff
@@ -43,19 +43,13 @@ Slot class >> checkValidName: aSymbol [
...
+	(#(#true #false #nil) includes: aSymbol) ifTrue: [
 		^ InvalidSlotName signalFor: aSymbol ].
 
-
 	(aSymbol allSatisfy: [ :aCharacter | aCharacter isAlphaNumeric or: [ aCharacter = $_ ] ]) ifFalse: [
 		^ InvalidSlotName signalFor: aSymbol ]
 ]
 
-{ #category : #testing }
-Slot class >> isPseudovariableName: aSymbol [
```
Comment (privat) [inline src/Kernel/Slot.class.st:55]: Keeping a list of named literals into an autonomous method is ok, I think.
Maybe a method of Symbol?

**Review by privat — APPROVED** [review]
#13136 is still not fixed. I'm all OK to transform hard errors into warnings, but don't forget the warning part!

**Comment by privat** [comment]
The real controversy will be for named literal, allowing an ivar named `true` could be a lot of fun.

**Comment by MarcusDenker** [comment]
> The real controversy will be for named literal, allowing an ivar named `true` could be a lot of fun.

I think on the lowest level, we should allow as much as possible. And then on the tool level warn and discourage very odd uses and guide users to not by mistake do strange things

File: src/Kernel/Slot.class.st (hunk 1 above)
Comment (MarcusDenker) [reply src/Kernel/Slot.class.st:55]: I have just removed the check. This will allow "nil true false", but the will never be used, thus this is save to do

**Comment by privat** [comment]
> I think on the lowest level, we should allow as much as possible. And then on the tool level warn and discourage very odd uses and guide users to not by mistake do strange things.

I agree, but for the specific case of named literal, there is an additional scanner/parser issue as nil true and false are not recognized as variables but as literal.
So currently. An ivar named `#true` can only access trough low-level VM stuff, high-level MOP stuff, synthetic AST, or custom scanner/parser.
But it's also an issue for any variables with a weird name, e.g. `#'1e ☭'`.

The named literal thing seems only important in literal arrays `#(true) = {true}` and `#(true) ~= #(#true)` whereas other identifiers are symbols `#(self) = #(#self)`

**Pull request merged**